compiler/rustc_target/src/spec/targets/i686_unknown_linux_gnu.rs

```diff
@@ -1,7 +1,13 @@
-use crate::spec::{Cc, LinkerFlavor, Lld, SanitizerSet, StackProbeType, Target, base};
+use crate::spec::{Cc, LinkerFlavor, Lld, RustcAbi, SanitizerSet, StackProbeType, Target, base};
 
 pub(crate) fn target() -> Target {
     let mut base = base::linux_gnu::opts();
+    base.rustc_abi = Some(RustcAbi::X86Sse2);
+    // Dear distribution packager, if you are changing the base CPU model with the goal of removing
+    // the SSE2 requirement, make sure to also set the `rustc_abi` to `None` above or else the compiler
+    // will complain that the chosen ABI cannot be realized with the given CPU features.
+    // Also note that x86 without SSE2 is *not* considered a Tier 1 target by the Rust project, and
+    // it has some known floating-point correctness issues that can lead to unsoundness.
```

Comment:
"unsoundness" sounds vague, and the alternative can also be "unsoundness" as seen by a distro packager (i.e. programs that incorporate Rust crashing due to SIGILL on CPUs they used to run on... usually a bad sign!)

I think we can specify the double-rounding issue and sometimes value truncation, because LLVM can incorrectly handle spilling a value to the stack and then pulling it back into an FP register? So float-using code may now have corrupted values, which may lead to incorrect math or memory unsoundness if code uses a float as a key to a data structure.

Reply:
I extended the comment.